runtime/sema/account.cdc

```diff
@@ -0,0 +1,521 @@
+access(all)
+struct Account {
+
+    /// The address of the account.
+    access(all)
+    let address: Address
+
+    /// The FLOW balance of the default vault of this account.
+    access(all)
+    let balance: UFix64
+
+    /// The FLOW balance of the default vault of this account that is available to be moved.
+    access(all)
+    let availableBalance: UFix64
+
+    /// The storage of the account.
+    access(AccountMapping)
+    let storage: Account.Storage
+
+    /// The contracts deployed to the account.
+    access(AccountMapping)
+    let contracts: Account.Contracts
+
+    /// The keys assigned to the account.
+    access(AccountMapping)
+    let keys: Account.Keys
+
+    /// The inbox allows bootstrapping (sending and receiving) capabilities.
+    access(AccountMapping)
+    let inbox: Account.Inbox
+
+    /// The capabilities of the account.
+    access(AccountMapping)
+    let capabilities: Account.Capabilities
```

Comment:
> It would be nice if there was a way for a field to restrict access to a field using an entitlement

I like the fact that entitlement-mapped fields reflect the reality of on-chain data in some ways; everything is publicly readable, no matter what level of access you have, so I think it makes sense to not be able to use maps to "pretend" like a field can be made unreadable when that is not the reality.

Reply:
Took some notes for potential improvements to the entitlements feature and documentation.